importers/src/bible/osis/index.ts

```diff
@@ -0,0 +1,918 @@
+                const currentContainer = this.getCurrentContainer(context);
+                if (currentContainer.type === 'section') {
+                    if (DEBUG_OUTPUT_ENABLED && DEBUG_LEVEL === 'info')
+                        console.log(
+                            this.getErrorMessageWithContext(
+                                'saving note as section description',
+                                context
+                            )
+                        );
+                    currentContainer.description = content;
+                } else {
+                    const currentPhrase = this.getCurrentPhrase(context, true);
+                    if (!currentPhrase) {
+                        if (DEBUG_OUTPUT_ENABLED) console.log(context);
+                        throw new Error(`note without a phrase`);
+                    }
+                    const note: IBibleNote = {
+                        key: tag.attributes.n,
+                        content
+                    };
+                    currentPhrase.notes = [note];
+                }
+                context.contentContainerStack.push(content);
+                break;
+            }
+            case 'crossReference': {
+                if (context.crossRefBuffer) {
+                    throw new Error(
+                        this.getErrorMessageWithContext(
+                            `cross reference buffer was not cleared`,
+                            context
+                        )
+                    );
+                }
+                context.crossRefBuffer = {
+                    key: tag.attributes.n,
+                    refs: []
+                };
+                break;
+            }
+            case 'catchWord':
+                // catchWord is handled in parseTextNode
+                break;
+            case 'title':
+                if (tag.attributes.canonical === 'true') {
+                    const titleGroup: IBibleContentGroup<'title'> = {
+                        type: 'group',
+                        groupType: 'title',
+                        contents: []
+                    };
+                    const currentContainer = this.getCurrentContainer(context);
+                    if (
+                        currentContainer.type !== 'group' ||
+                        currentContainer.groupType !== 'paragraph'
+                    ) {
+                        const paragraph: IBibleContentGroup<'paragraph'> = {
+                            type: 'group',
+                            groupType: 'paragraph',
+                            contents: [titleGroup]
+                        };
+
+                        currentContainer.contents.push(paragraph);
+                    } else currentContainer.contents.push(titleGroup);
+                    context.contentContainerStack.push(titleGroup);
+                }
+                // section title is handled in parseTextNode
+                break;
+            case 'w':
```

Comment:
because some osis tags like `'w'` aren't super descriptive, helpful to have an OsisTags enum

Reply:
i put your enums to a shared file, added some missing types and migrated the osis-importer to using it